packages/core/utils/src/traverse/query-populate.ts

```diff
@@ -27,15 +26,37 @@ const isKeyword = (keyword: string) => {
 const isStringArray = (value: unknown): value is string[] =>
   isArray(value) && value.every(isString);
 
+const isWildCardConstant = (value: unknown): value is '*' => value === '*';
+
 const isObj = (value: unknown): value is Record<string, unknown> => isObject(value);
 
 const populate = traverseFactory()
   // Array of strings ['foo', 'foo.bar'] => map(recurse), then filter out empty items
   .intercept(isStringArray, async (visitor, options, populate, { recurse }) => {
-    return Promise.all(populate.map((nestedPopulate) => recurse(visitor, options, nestedPopulate)));
+    const visitedPopulate = await Promise.all(
+      populate.map((nestedPopulate) => recurse(visitor, options, nestedPopulate))
+    );
+
+    return visitedPopulate.filter((item) => !isNil(item));
+  })
+  // Transform wildcard populate to an exhaustive list of attributes to populate.
+  .intercept(isWildCardConstant, (visitor, options, _data, { recurse }) => {
+    const attributes = options.schema?.attributes;
+
+    // This should never happen, but adding the check in
+    // case this method is called with wrong parameters
+    if (!attributes) {
+      return '*';
+    }
+
+    const parsedPopulate = Object.entries(attributes)
+      // Get the list of all attributes that can be populated
+      .filter(([, value]) => ['relation', 'component', 'dynamiczone', 'media'].includes(value.type))
+      // Only keep the attributes key
+      .map(([key]) => key);
```

Comment:
replace by { [key]: true} to avoid adding the new array feature

Reply:
:+1: